controllers/machineset_controller.go

```diff
@@ -250,28 +258,21 @@ func (r *MachineSetReconciler) reconcile(ctx context.Context, cluster *clusterv1
 
 	syncErr := r.syncReplicas(ctx, machineSet, filteredMachines)
 
-	ms := machineSet.DeepCopy()
-	newStatus, err := r.calculateStatus(ctx, cluster, ms, filteredMachines)
-	if err != nil {
-		return ctrl.Result{}, errors.Wrapf(err, "failed to calculate MachineSet's Status")
-	}
-
 	// Always updates status as machines come up or die.
-	updatedMS, err := r.patchMachineSetStatus(ctx, machineSet, newStatus)
-	if err != nil {
+	if err = r.updateStatus(ctx, cluster, machineSet, filteredMachines); err != nil {
 		if syncErr != nil {
-			return ctrl.Result{}, errors.Wrapf(err, "failed to sync machines: %v. failed to patch MachineSet's Status", syncErr)
+			return ctrl.Result{}, errors.Wrapf(err, "failed to sync machines: %v. failed to update MachineSet's status", syncErr)
 		}
-		return ctrl.Result{}, errors.Wrap(err, "failed to patch MachineSet's Status")
+		return ctrl.Result{}, errors.Wrapf(err, "failed to update MachineSet's Status")
```

Comment:
```suggestion
		return ctrl.Result{}, errors.Wrapf(kerrors.NewAggregate([]errors{err, syncErr}), "failed to update MachineSet's Status")
```

And we can remove the inner `if syncErr != nil`